Improvement

chcmedeiros opened the pull request.

chcmedeiros committed:
Adds check for excessive number of items

Adds a check to prevent transactions from being processed if they contain an excessive amount of items, which could lead to display issues.

chcmedeiros committed:
Increases patch version and updates Zemu snapshots

Increases the patch version in Makefile.version from 4 to 5, indicating a new patch release.

Updates Zemu snapshots to reflect the changes, ensuring the UI tests remain accurate.
This includes renaming some snapshots to align with expected naming conventions.

0xPxt commented on app/src/tx_display.c:
```diff
@@ -473,14 +473,20 @@ parser_error_t tx_display_numItems(uint8_t *num_items) {
+  }
+
+  // Reject transactions with too many items to display safely
+  if (total > UINT8_MAX) {
```
Would it be better to fix the overflows in `get_subitem_count` directly? The function expects a pointer to a `uint8_t` so I think it makes sense to handle those errors there. What do you think?

chcmedeiros committed:
Validates subitem count to prevent overflow

Adds a check to ensure the calculated number of subitems does not exceed the maximum value of a uint8_t.
This prevents potential integer overflows when casting to uint8_t, enhancing the application's robustness.
